pkg/metamask/abi/fourbyte/type.go

```diff
@@ -23,6 +23,7 @@ const (
 
 // Type is the reflection of the supported argument type.
 type Type struct {
+	// TODO change type of elem to `Argument`
```

Comment:
Why? I think `Type` structure contains all necessary information in case of `SliceTy`.